components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -2835,7 +2835,7 @@ private static void addAPIObj(SubscribedAPI subscribedAPI, NativeArray apisArray
             apiObj.put("hasMultipleEndpoints", apiObj, String.valueOf(api.getSandboxUrl() != null));
             apisArray.put(apisArray.getIds().length, apisArray, apiObj);
         } catch (APIManagementException e) {
-            handleException("Error while obtaining application metadata", e);
+            log.error("Error while obtaining application metadata", e);
```

Comment:
Any reason for just log the exception as it was previously logged and throw the exception with handleException?